pdoc/doc.py

```diff
@@ -319,13 +321,24 @@ def members(self) -> dict[str, Doc]:
                     qualname,
                     docstring="",
                     annotation=self._var_annotations.get(name, empty),
-                    default_value=obj,
+                    default_value=_pydantic.default_value(self.obj, name, obj),
                     taken_from=taken_from,
                 )
-            if self._var_docstrings.get(name):
-                doc.docstring = self._var_docstrings[name]
-            if self._func_docstrings.get(name) and not doc.docstring:
-                doc.docstring = self._func_docstrings[name]
+
+            doc.docstring = next(
+                (
+                    d
+                    for d in [
+                        _pydantic.get_field_docstring(cast(type, self.obj), name),
+                        self._var_docstrings.get(name),
+                        doc.docstring,
+                        self._func_docstrings.get(name),
+                    ]
+                    if d
+                ),
+                doc.docstring,
+            )
```

Comment:
This code is too smart. Simple if statements are much easier to read.

Reply:
Kinda suspected you would say that. Will revise.